value/src/org/immutables/value/Value.java

```diff
@@ -1149,6 +1149,93 @@
     String redactedMask() default "";
 
     /**
+     * When enabled: immutable attributes with discoverable builders will receive the additional
+     * builder API:
+     *
+     * For single children:
+     * {@code get*Builder()}
+     * {@code set*Builder(BuilderT builder)}
+     *
+     * For collection of children:
+     * {@code add*Builder()}
+     * {@code addAll*Builder(Iterable<BuilderT> builderCollection)}
+     * {@code addAll*Builder(BuilderT... builderArgs)}
+     * {@code get*BuilderList()}
+     *
+     * In strict mode, you may only call set*Builder once, but you may call get*Builder multiple
+     * times, in which the same builder is returned. If the nested immutable is also strict,
+     * then you will only be able to set properties on the child builder once.
+     *
+     * To discover builders on value attributes the value methods are scanned for method names
+     * matching a patterns specified in {@link #nestedBuilderDetectionPattern()}.
+     *
+     * This style parameter is experimental and may change in future.
+     * @return true to enable the feature.
+     */
+    boolean nestedBuilderDetection() default false;
```

Comment:
This popped up from the top of my head just now: we could also apply "attributeBuilder" term for the functionality. As I also like `nestedBuilder` as it is here, I cannot decide one over another, so perhaps it's better to stick to "nested"